account-kit/core/src/store/store.ts

```diff
@@ -265,7 +275,16 @@ const addClientSideStoreListeners = (store: Store) => {
     (signer) => {
       if (!signer) return;
       signer.on("statusChanged", (status) => {
-        store.setState({ signerStatus: convertSignerStatusToState(status) });
+        const shouldClearError =
+          status === AlchemySignerStatus.AUTHENTICATING ||
+          status === AlchemySignerStatus.AWAITING_EMAIL_AUTH ||
+          status === AlchemySignerStatus.CONNECTED;
+        store.setState((state) => ({
+          signerStatus: convertSignerStatusToState(
+            status,
+            shouldClearError ? undefined : state.signerStatus.error
+          ),
+        }));
```

Comment:
can we add tests for this in `store.test.ts`?

Reply:
Simplified the logic so it shouldn't need more testing than the rest of the listeners now.